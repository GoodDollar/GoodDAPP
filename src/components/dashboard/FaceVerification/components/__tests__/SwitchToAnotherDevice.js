import React from 'react'
import renderer from 'react-test-renderer'
import { noop } from 'lodash'

import ImportedSwitchToAnotherDevice from '../SwitchToAnotherDevice'
import { withThemeProvider } from '../../../../../__tests__/__util__'

describe('FaceVerification SwitchToAnotherDevice', () => {
  const SwitchToAnotherDevice = withThemeProvider(ImportedSwitchToAnotherDevice)
  const screenProps = { goToRoot: jest.fn(noop) }

  afterEach(() => {
    screenProps.goToRoot.mockReset()
  })

  it('renders without errors', () => {
    let tree

    expect(() => (tree = renderer.create(<SwitchToAnotherDevice screenProps={screenProps} />))).not.toThrow()
    expect(tree.toJSON()).toBeTruthy()
  })

  it('matches snapshot', () => {
    const tree = renderer.create(<SwitchToAnotherDevice screenProps={screenProps} />)

    expect(tree.toJSON()).toMatchSnapshot()
  })

  it('should execute screenProps.goToRoot on "OK" press', () => {
    const tree = renderer.create(<SwitchToAnotherDevice screenProps={screenProps} />)
    const button = tree.root.findByProps({ testID: 'ok_button' })

    button.props.onPress()
    expect(screenProps.goToRoot).toHaveBeenCalled()
  })
})
