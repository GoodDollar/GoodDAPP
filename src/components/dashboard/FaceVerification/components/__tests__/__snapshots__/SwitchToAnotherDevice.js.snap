// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FaceVerification SwitchToAnotherDevice matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "WebkitBoxFlex": 1,
      "WebkitFlexBasis": "0%",
      "WebkitFlexGrow": 1,
      "WebkitFlexShrink": 1,
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
      "msFlexNegative": 1,
      "msFlexPositive": 1,
      "msFlexPreferredSize": "0%",
    }
  }
>
  <div
    className="css-view-1dbjc4n"
    data-name="viewWrapper"
    style={
      Object {
        "WebkitBoxDirection": "normal",
        "WebkitBoxFlex": 1,
        "WebkitBoxOrient": "vertical",
        "WebkitFlexDirection": "column",
        "WebkitFlexGrow": 1,
        "backgroundImage": "linear-gradient(to bottom, #00AFFF, #2DC0F7, #28C0EF, #23C0E7, #1EC1DF, #19C1D7, #14C1CF, #0FC2C7, #0FC2C7, #0AC2BF, #05C2B7, #00C3AF)",
        "display": "flex",
        "flexDirection": "column",
        "flexGrow": 1,
        "maxHeight": "844px",
        "msFlexDirection": "column",
        "msFlexPositive": 1,
        "paddingBottom": "8px",
        "paddingLeft": "8px",
        "paddingRight": "8px",
        "paddingTop": "8px",
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "WebkitAlignItems": "center",
          "WebkitBoxAlign": "center",
          "WebkitBoxFlex": 1,
          "WebkitFlexBasis": "0%",
          "WebkitFlexGrow": 1,
          "WebkitFlexShrink": 0,
          "alignItems": "center",
          "backgroundColor": "rgba(255,255,255,1.00)",
          "borderBottomLeftRadius": "5px",
          "borderBottomRightRadius": "5px",
          "borderTopLeftRadius": "5px",
          "borderTopRightRadius": "5px",
          "display": "flex",
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 0,
          "height": "100%",
          "msFlexAlign": "center",
          "msFlexNegative": 0,
          "msFlexPositive": 1,
          "msFlexPreferredSize": "0%",
          "paddingBottom": "16px",
          "paddingTop": "16px",
        }
      }
    >
      <div
        className="css-view-1dbjc4n"
        style={
          Object {
            "WebkitBoxFlex": 1,
            "WebkitBoxPack": "justify",
            "WebkitFlexBasis": "0%",
            "WebkitFlexGrow": 1,
            "WebkitFlexShrink": 1,
            "WebkitJustifyContent": "space-between",
            "backgroundColor": "rgba(255,255,255,1.00)",
            "borderBottomLeftRadius": "5px",
            "borderBottomRightRadius": "5px",
            "borderTopLeftRadius": "5px",
            "borderTopRightRadius": "5px",
            "flexBasis": "0%",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "space-between",
            "marginBottom": "0px",
            "msFlexNegative": 1,
            "msFlexPack": "justify",
            "msFlexPositive": 1,
            "msFlexPreferredSize": "0%",
            "paddingBottom": "44px",
            "paddingLeft": "32px",
            "paddingRight": "32px",
            "paddingTop": "22px",
            "width": "100%",
          }
        }
      >
        <div
          className="css-text-901oao"
          dir="auto"
          style={
            Object {
              "color": "rgba(250,108,119,1.00)",
              "direction": "ltr",
              "fontFamily": "Roboto",
              "fontSize": "24px",
              "fontWeight": "500",
              "lineHeight": "30px",
              "marginBottom": "8px",
              "marginTop": "8px",
              "textAlign": "center",
              "textDecoration": "none",
              "textTransform": "none",
            }
          }
        >
          ,
Please try to switch
to another device
        </div>
        <div
          className="css-view-1dbjc4n"
          style={
            Object {
              "WebkitBoxFlex": 1,
              "WebkitBoxPack": "center",
              "WebkitFlexBasis": "auto",
              "WebkitFlexGrow": 1,
              "WebkitJustifyContent": "center",
              "flexBasis": "auto",
              "flexGrow": 1,
              "height": "182.02597402597402px",
              "justifyContent": "center",
              "msFlexPack": "center",
              "msFlexPositive": 1,
              "msFlexPreferredSize": "auto",
              "overflowX": "hidden",
              "overflowY": "hidden",
              "zIndex": 0,
            }
          }
        >
          <div
            className="css-view-1dbjc4n"
            style={
              Object {
                "backgroundColor": "rgba(0,0,0,0.00)",
                "backgroundImage": "url(\\"FRSwitchToAnotherDevice.svg\\")",
                "backgroundPosition": "center",
                "backgroundRepeat": "no-repeat",
                "backgroundSize": "contain",
                "bottom": "0px",
                "height": "100%",
                "left": "0px",
                "position": "absolute",
                "right": "0px",
                "top": "0px",
                "width": "100%",
                "zIndex": -1,
              }
            }
          />
          <img
            alt=""
            className="css-accessibilityImage-9pa8cd"
            draggable={false}
            src="FRSwitchToAnotherDevice.svg"
          />
        </div>
        <div
          className="css-view-1dbjc4n"
          style={
            Object {
              "WebkitBoxFlex": 1,
              "WebkitBoxPack": "center",
              "WebkitFlexGrow": 1,
              "WebkitJustifyContent": "center",
              "backgroundColor": "rgba(255,255,255,1.00)",
              "borderBottomLeftRadius": "5px",
              "borderBottomRightRadius": "5px",
              "borderTopLeftRadius": "5px",
              "borderTopRightRadius": "5px",
              "flexGrow": 1,
              "justifyContent": "center",
              "marginBottom": "0px",
              "msFlexPack": "center",
              "msFlexPositive": 1,
              "paddingBottom": "0px",
              "paddingLeft": "0px",
              "paddingRight": "0px",
              "paddingTop": "0px",
            }
          }
        >
          <div
            className="css-view-1dbjc4n"
            style={
              Object {
                "backgroundColor": "rgba(0,175,255,1.00)",
                "borderBottomLeftRadius": "1px",
                "borderBottomRightRadius": "1px",
                "borderTopLeftRadius": "1px",
                "borderTopRightRadius": "1px",
                "height": "2px",
              }
            }
          />
          <div
            className="css-view-1dbjc4n"
            style={
              Object {
                "paddingBottom": "16px",
                "paddingTop": "16px",
              }
            }
          >
            <div
              className="css-text-901oao"
              dir="auto"
              style={
                Object {
                  "color": "rgba(0,175,255,1.00)",
                  "direction": "ltr",
                  "fontFamily": "Roboto",
                  "fontSize": "18px",
                  "fontWeight": "normal",
                  "lineHeight": "25px",
                  "textAlign": "center",
                  "textDecoration": "none",
                  "textTransform": "none",
                }
              }
            >
              Sometimes, switching to a
different device is a good solution.
            </div>
            <div
              className="css-text-901oao"
              dir="auto"
              style={
                Object {
                  "color": "rgba(0,175,255,1.00)",
                  "direction": "ltr",
                  "fontFamily": "Roboto",
                  "fontSize": "18px",
                  "fontWeight": "700",
                  "lineHeight": "25px",
                  "textAlign": "center",
                  "textDecoration": "none",
                  "textTransform": "none",
                }
              }
            >
              Sorry about thatâ€¦ :)
            </div>
          </div>
          <div
            className="css-view-1dbjc4n"
            style={
              Object {
                "backgroundColor": "rgba(0,175,255,1.00)",
                "borderBottomLeftRadius": "1px",
                "borderBottomRightRadius": "1px",
                "borderTopLeftRadius": "1px",
                "borderTopRightRadius": "1px",
                "height": "2px",
              }
            }
          />
        </div>
      </div>
      <div
        className="css-view-1dbjc4n"
        style={
          Object {
            "paddingLeft": "8px",
            "paddingRight": "8px",
            "width": "100%",
          }
        }
      >
        <div
          className="css-view-1dbjc4n"
          data-testid="ok_button"
          style={
            Object {
              "WebkitAlignItems": "stretch",
              "WebkitBoxAlign": "stretch",
              "WebkitBoxPack": "center",
              "WebkitJustifyContent": "center",
              "alignItems": "stretch",
              "backgroundColor": "rgba(0,175,255,1.00)",
              "borderBottomColor": "rgba(0,175,255,1.00)",
              "borderBottomLeftRadius": "50px",
              "borderBottomRightRadius": "50px",
              "borderBottomStyle": "solid",
              "borderBottomWidth": "0px",
              "borderLeftColor": "rgba(0,175,255,1.00)",
              "borderLeftStyle": "solid",
              "borderLeftWidth": "0px",
              "borderRightColor": "rgba(0,175,255,1.00)",
              "borderRightStyle": "solid",
              "borderRightWidth": "0px",
              "borderTopColor": "rgba(0,175,255,1.00)",
              "borderTopLeftRadius": "50px",
              "borderTopRightRadius": "50px",
              "borderTopStyle": "solid",
              "borderTopWidth": "0px",
              "boxShadow": "0px 0.75px 1.5px rgba(0,0,0,0.24)",
              "display": "flex",
              "justifyContent": "center",
              "marginBottom": "0px",
              "marginLeft": "0px",
              "marginRight": "0px",
              "marginTop": "0px",
              "minHeight": "44px",
              "minWidth": "64px",
              "msFlexAlign": "stretch",
              "msFlexPack": "center",
              "paddingBottom": "0px",
              "paddingLeft": "0px",
              "paddingRight": "0px",
              "paddingTop": "0px",
            }
          }
        >
          <div
            className="css-cursor-18t94o4 css-view-1dbjc4n"
            data-focusable={true}
            disabled={false}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onKeyUp={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            role="button"
            style={
              Object {
                "borderBottomLeftRadius": "50px",
                "borderBottomRightRadius": "50px",
                "borderTopLeftRadius": "50px",
                "borderTopRightRadius": "50px",
                "cursor": "pointer",
                "msTouchAction": "manipulation",
                "overflowX": "hidden",
                "overflowY": "hidden",
                "position": "relative",
                "touchAction": "manipulation",
              }
            }
            tabIndex="0"
          >
            <div
              className="css-view-1dbjc4n"
              style={
                Object {
                  "WebkitAlignItems": "center",
                  "WebkitBoxAlign": "center",
                  "WebkitBoxDirection": "normal",
                  "WebkitBoxOrient": "horizontal",
                  "WebkitBoxPack": "center",
                  "WebkitFlexDirection": "row",
                  "WebkitJustifyContent": "center",
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "msFlexAlign": "center",
                  "msFlexDirection": "row",
                  "msFlexPack": "center",
                }
              }
            >
              <div
                className="css-text-901oao css-textOneLine-bfa6kz"
                dir="auto"
                style={
                  Object {
                    "color": "rgba(255,255,255,1.00)",
                    "direction": "ltr",
                    "fontFamily": "Roboto-Medium, Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                    "letterSpacing": "1px",
                    "marginBottom": "9px",
                    "marginLeft": "16px",
                    "marginRight": "16px",
                    "marginTop": "9px",
                    "textAlign": "center",
                  }
                }
              >
                <div
                  className="css-view-1dbjc4n"
                  style={
                    Object {
                      "WebkitAlignItems": "center",
                      "WebkitBoxAlign": "center",
                      "WebkitBoxDirection": "normal",
                      "WebkitBoxFlex": 1,
                      "WebkitBoxOrient": "horizontal",
                      "WebkitFlexBasis": "0%",
                      "WebkitFlexDirection": "row",
                      "WebkitFlexGrow": 1,
                      "WebkitFlexShrink": 1,
                      "alignItems": "center",
                      "display": "flex",
                      "flexBasis": "0%",
                      "flexDirection": "row",
                      "flexGrow": 1,
                      "flexShrink": 1,
                      "msFlexAlign": "center",
                      "msFlexDirection": "row",
                      "msFlexNegative": 1,
                      "msFlexPositive": 1,
                      "msFlexPreferredSize": "0%",
                    }
                  }
                >
                  <span
                    className="css-text-901oao css-textHasAncestor-16my406"
                    dir="auto"
                    style={
                      Object {
                        "WebkitAlignItems": "center",
                        "WebkitBoxAlign": "center",
                        "WebkitBoxPack": "center",
                        "WebkitJustifyContent": "center",
                        "alignItems": "center",
                        "color": "rgba(255,255,255,1.00)",
                        "direction": "ltr",
                        "fontFamily": "Roboto",
                        "fontSize": "16px",
                        "fontWeight": "500",
                        "justifyContent": "center",
                        "letterSpacing": "0px",
                        "lineHeight": "22px",
                        "msFlexAlign": "center",
                        "msFlexPack": "center",
                        "paddingTop": "1px",
                        "textAlign": "center",
                        "textDecoration": "none",
                        "textTransform": "uppercase",
                      }
                    }
                  >
                    OK
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
